### PR TITLE
adding overlay filter and tests

### PR DESCRIPTION
**Pull request by rgritti**

### CODE REVIEW

File: filters/overlayImage.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	var x, y int
+	switch r.verticalGravity {
+	case bimg.GravityNorth:
+		y = r.topMargin
+	case bimg.GravityCentre:
+		y = r.topMargin + int(float64(origSize.Height-r.topMargin-r.bottomMargin)/2) - int(float64(overSize.Height)/2)
```
Comment (danpersa) [inline filters/overlayImage.go]: did you run `go format`? this line looks weird...

File: filters/overlayImage.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+	return &bimg.Options{WatermarkImage: bimg.WatermarkImage{Buf: overArr,
+		Opacity: float32(r.opacity),
+		Left:    x,
+		Top:     y,
+	}}, nil
+}
+
+func readImage(file string) []byte {
+	img, _ := os.Open(file)
```
Comment (chandu188) [inline filters/overlayImage.go]: Should n't we handle error case here?

File: filters/overlayImage.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	switch r.horizontalGravity {
+	case bimg.GravityWest:
+		x = r.leftMargin
+	case bimg.GravityCentre:
+		x = r.leftMargin + int(float64(origSize.Width-r.leftMargin-r.rightMargin)/2) - int(float64(overSize.Width)/2)
+	case bimg.GravityEast:
+		x = origSize.Width - r.rightMargin - overSize.Width
+	}
+
+	// in case y overflows the image
```
Comment (chandu188) [inline filters/overlayImage.go]: maybe you meant 'x' here in the comment

File: filters/overlayImage.go (hunk 1 above)
Comment (rgritti) [inline filters/overlayImage.go]: Yes, this is the standard formatting...

Comment (rgritti) [inline filters/overlayImage.go]: every time I see this in the IDE... I indent the 'case'... and then it is reverted back :(

**Pull request merged**